src/main/riddl/Events/eventContext.riddl

```
@@ -0,0 +1,197 @@
+                on command StartEvent{
+                  then error "Event is already started. Cannot start an event that is already in progress"
+                }
+                on command EndEvent{
+									then "morph Event entity to EventStatus.Past"
+                  and "set entity lastModifiedBy to Event.endingMember"
+                  and "set entity lastModifiedOn to today()"
+                }
+                on command AddLiveUpdate{???}
+            }
+            //score, period, etc...
+        }
+        state Delayed is {
+            fields {
+                eventId: EventId,
+                eventInfo: EventInfo,
+                reason: String,
+                eventMetaInfo: EventMetaInfo
+                //finalScore
+            }
+            handler DelayedEventHandler is {
+                on command ScheduleEvent{
+									then "morph Event entity to EventStatus.InProgress"
+                  and "set entity lastModifiedBy to Event.delayingMember"
+                  and "set entity lastModifiedOn to today()"
+                }
+                on command ChangeEventInfo{
+                  then "set entity eventInfo to message info"
+                  and "set entity lastModifiedBy to Event.cancellingMember"
+                  and "set entity lastModifiedOn to today()"
+                }
+                on command CancelEvent{
+									then "morph Event entity to EventStatus.Cancelled"
+                  and "set entity lastModifiedBy to Event.cancellingMember"
+                  and "set entity lastModifiedOn to today()"
+                }
+                on command RescheduleEvent{
+                  then "set entity evenInfo.expectedStart to Event.start"
+                  and "set entity evenInfo.expectedEnd to Event.end"
+                  and "set entity lastModifiedBy to Event.reschedulingMember"
+                  and "set entity lastModifiedOn to today()"
+                }
+                on command DelayEvent{
```

Because we are already in a delayed state I think this should throw an error.